src/wobbly/dependencies/search.py

```diff
@@ -0,0 +1,51 @@
+"""Dependency collecting common parameters for a job search."""
+
+from typing import Annotated
+
+from fastapi import Query
+from safir.pydantic import UtcDatetime
+from vo_models.uws.types import ExecutionPhase
+
+from ..models import JobCursor, JobSearch
+
+__all__ = ["job_search_dependency"]
+
+
+async def job_search_dependency(
+    *,
+    phase: Annotated[
+        list[ExecutionPhase] | None,
+        Query(
+            title="Execution phase",
+            description="Limit results to the provided execution phases",
+        ),
+    ] = None,
+    since: Annotated[
+        UtcDatetime | None,
+        Query(
+            title="Creation date",
+            description="Limit results to jobs created after this date",
+        ),
+    ] = None,
+    cursor: Annotated[
+        str | None,
+        Query(
+            title="Pagination cursor",
+            description="Cursor used when moving between pages of results",
+        ),
+    ] = None,
+    limit: Annotated[
+        int | None,
+        Query(
+            title="Number of jobs",
+            description="Return at most the given number of jobs",
+        ),
```

Comment:
`ge=1` is a great idea, added. Since we allow not setting the limit at all, I decided that having an upper limit didn't make sense. We can cap the limit and force pagination later if that ends up being a problem, but since the common case will be passing along a UWS query, and UWS currently doesn't support pagination, we're stuck with the memory usage hit in the service anyway and I'm not sure it's worth restricting it here.